### ISSUE TITLE
DeleteRow in Tabulator.js not working anymore with new version 4.4

### ISSUE THREAD

**Issue by RegCoding** [olifolkerd/tabulator#2368]
Dear Oli,

We are currently using following javascript code to delete selected rows from a tabulator.js table. We now switched to tabulator.js version 4.4 and now the code is not working anymore. This is the javascript code. The focus is on the two lines with deleteRow and redraw, which are not working anymore after switching to version 4.4:

<script language="JavaScript" type="text/javascript">
        $("#impact_delete_<%= OBJ_ID %>_bestaetigung").click(function() {
                $.post("saveNotRelevant_verlinkungen.ims", 
                {
                    cache: false,
                    contentType: 'application/x-www-form-urlencoded;     charset=UTF-8',
                    OBJ_ID: '<%= OBJ_ID %>',
                    INV_ID: '<%= R_OBJ_ID %>',
                    R_OBJ_ID: '<%= R_OBJ_ID %>',
                    aktion: 'einblenden'
                },
                function(data) {            
                    $('#noteIM-SUITE').html(data);      
                    $("#tabulator_verlinkungen").tabulator("deleteRow", <%=     deleteZeileInt %>); 
                    $("#tabulator_verlinkungen").tabulator("redraw", true);
                    $("#dialogView_1").dialog('close');
                });
        });
</script>

I already changed the code to $("#tabulator_verlinkungen").deleteRow(<%= deleteZeileInt %>);
but this is also not working.

The javascript function should delete one row with id <%= deleteZeileInt %> of the tabulator.js table (the table is saved on $("#tabulator_verlinkungen") and redraw the table.

I am really looking forward to your help. 

Best regards, 
Gerrit

**Comment by tentus** [comment]
What was your previous version? Can you post your constructor? As is I'm not sure if you're upgrading from version 3, if you're using the jquery wrapper, etc etc.

**Comment by RegCoding** [comment]
Hi, thank you for your response and your support.
We are currently using version 4.4 coming from version 4.1.4. We use jquery_wrapper.min.js.
The issue in detail: From the tabulator.js table we call a modal JQuery dialog to let the user confirm the deletion of the row. Then we delete the row from js of the modal dialog with the js that I have post above. With version 4.1.4 this worked well, but now the execution of line $('#tabulator_verlinkungen').deleteRow( 5) in this js causes that all remaining code lines in the main js of the tabulator.js table (from which we called the modal dialog) will not be executed anymore.
Here you find two screenshot of the table and the modal dialog. and also the js-File of the table from which we call the modal dialog and which remaining lines of code will not be executed anymore after we have deleted the row.
Do you know what changed from version 4.1.4 to version 4.4 what causes this behaviour and how we could fix that?
![Bildschirmfoto 2019-09-23 um 21 40 35](https://user-images.githubusercontent.com/44733616/65500335-7126d180-debf-11e9-84fd-c97af2621fac.png)

[tabulatorjs.txt](https://github.com/olifolkerd/tabulator/files/3646524/tabulatorjs.txt)

**Comment by tentus** [comment]
There was a small change to how the rerendering is done back in 4.2, by adding a `blockRedraw` param to `deleteRow`, but the default behavior should have the same. 

Are you getting any errors in the console? If so, you would copy-paste them here?

**Comment by RegCoding** [comment]
No, we are getting no errors. The code just stops to execute after the row has been deleted. I have attached here the code of the js that is been executed from the modal dialog to delete the row. We have this kind of feature (deleting a row from tabulator table after confirming from a modal dialog) at various places of our solution. 
Can you see if there is a mistake in the attached js?

**Comment by RegCoding** [comment]
[deleteRow.txt](https://github.com/olifolkerd/tabulator/files/3656288/deleteRow.txt)

**Comment by olifolkerd** [comment]
Hey @RegCoding 

Could you create a JS Fiddle that demonstrates your table in action, as a general rule i dont download users source for diagnostics.

If you take a look at the [Add/Delete Row Example](http://tabulator.info/examples/4.4#adddel) in the documentation you can see the functionality working in 4.4

Once we can see the table in action in a JSFiddle we should be able to help you out.

Cheers

Oli :)

**Comment by RegCoding** [comment]
Hi Oli,

thank you for your response. I build a basic scenario in jsfiddle to emphazise what we want to achieve:
https://jsfiddle.net/gknippschild/gzh6ys1d/24/

The problem I indicated you can find in row 9 of the js-section. In the jsfiddle scenario I could easily call table.deleteRow(1); instead, which would work immediately.
But in our real scenario JQuery dialog and tabulator.js initialization takes places in two different JSPs, so that the JQuery dialog would not know the local variable "table", which holds the Tabulator information.
This is why in our real scenario we try to delete the row with
$("#link_table").deleteRow(<%= row_number %>);

With Tabulator.js 4.1.4this approach works well, but with Tabulator 4.4 the row is not deleted and the following lines of code not executed.

**Comment by takuy** [comment]
@RegCoding, maybe this'll be a better solution:
https://jsfiddle.net/jeLy5f3h/1/

When you click on a row, you set data for the row-id on the dialog element. Then, you use that data to target a specific row for deletion.
